core/src/avm1/globals/movie_clip_loader.rs

```diff
@@ -42,91 +42,116 @@ pub fn load_clip<'gc>(
     this: Object<'gc>,
     args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
-    let url_val = args.get(0).cloned().unwrap_or(Value::Undefined);
-    let url = url_val.coerce_to_string(activation)?;
-    let target = args.get(1).cloned().unwrap_or(Value::Undefined);
-
-    if let Value::Object(target) = target {
-        if let Some(mc) = target
-            .as_display_object()
-            .and_then(|dobj| dobj.as_movie_clip())
-        {
-            let fetch = activation
-                .context
-                .navigator
-                .fetch(&url.to_utf8_lossy(), RequestOptions::get());
-            let process = activation.context.load_manager.load_movie_into_clip(
-                activation.context.player.clone().unwrap(),
-                DisplayObject::MovieClip(mc),
-                fetch,
-                url.to_string(),
-                None,
-                Some(this),
-            );
-
-            activation.context.navigator.spawn_future(process);
+    if let [url, target, ..] = args {
+        if let Value::String(url) = url {
+            let target = match target {
+                Value::String(_) => {
+                    let start_clip = activation.target_clip_or_root()?;
+                    activation.resolve_target_display_object(start_clip, *target, true)?
+                }
+                Value::Number(level_id) => {
+                    // Levels are rounded down.
+                    // TODO: What happens with negative levels?
+                    Some(activation.resolve_level(*level_id as i32))
+                }
+                Value::Object(object) => object.as_display_object(),
+                _ => None,
+            };
```

Comment:
Maybe extract this logic into a fn since it's used by several of the methods?

Reply:
Unfortunately, they have subtle differences:
* `load_clip` uses `resolve_level`.
* `unload_clip` and `get_progress` use `child_by_depth`
* `get_progress` returns `undefined` for non-DisplayObject objects.

It might be possible, but I doubt how better it will be.